client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -1044,26 +1053,53 @@ const getCompressParams = (
 export const compressAppSelector = (state) =>
   state.workbench.config.compressApp;
 
+export const systemsSelector = (state) => state.systems.storage.configuration;
+
 export function* compressFiles(action) {
-  const compressErrorAction = {
-    type: 'DATA_FILES_SET_OPERATION_STATUS',
-    payload: { status: 'ERROR', operation: 'compress' },
+  const compressErrorAction = (errorMessage) => {
+    return {
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: {
+        status: { type: 'ERROR', message: errorMessage },
+        operation: 'compress',
+      },
+    };
   };
+
   try {
     yield put({
       type: 'DATA_FILES_SET_OPERATION_STATUS',
-      payload: { status: 'RUNNING', operation: 'compress' },
+      payload: { status: { type: 'RUNNING' }, operation: 'compress' },
     });
     const compressApp = yield select(compressAppSelector);
     const defaultAllocation = yield select(defaultAllocationSelector);
     const latestCompress = yield call(fetchAppDefinitionUtil, compressApp);
+    const systems = yield select(systemsSelector);
+
+    let defaultPrivateSystem;
+
+    if (
+      action.payload.scheme !== 'private' &&
```

Comment:
nit: We have logic like this else where too, may be abstracting this into a util helps us understand the need for it, right now i can only read this is a condition with knowledge of reason behind it.
There are two usages like this else other.
`(s.scheme !== 'private' && s.scheme !== 'projects') `

Reply:
This can be done in a separate task

Reply:
Yes I agree that this would be confusing for another developer to look at without context. I can create a task to further explore this and potentially create a util function that can be used throughout

Reply:
Created Jira task [WP-263 ](https://jira.tacc.utexas.edu/browse/WP-263)for this